test/reply-error.test.js

```diff
@@ -369,3 +369,49 @@ test('Error.status property support', t => {
     )
   })
 })
+
+test('Support rejection with values that are not Error instances', t => {
+  const objs = [
+    0,
+    '',
+    [],
+    {},
+    null,
+    undefined,
+    123,
+    'abc',
+    new RegExp(),
+    new Date(),
+    new Uint8Array()
+  ]
+  for (const nonErr of objs) {
+    t.test('Type: ' + typeof nonErr, t => {
+      const fastify = Fastify()
+
+      fastify.get('/', () => {
+        return Promise.reject(nonErr)
+      })
+
+      fastify.inject({
+        method: 'GET',
+        url: '/'
+      }, res => {
+        t.strictEqual(res.statusCode, 500)
+        t.deepEqual(
+          JSON.parse(res.payload),
+          {
+            error: statusCodes['500'],
+            message: '',
+            statusCode: 500
+          }
+        )
+      })
+
+      fastify.setErrorHandler((err, reply) => {
+        t.strictEqual(err, nonErr)
+        t.end()
```

Comment:
The end of the test should be in the injection callback, not here

Reply:
This handler fires last. Not sure how I can end the test before it runs?

Reply:
How could it be possible? The inject callback has to be called after the setErrorHadler. @mcollina

Reply:
@allevo 😓 Upon closer inspection, it appears I still misunderstood tap's parallel subtests. I'll refactor again. WIP.